packages/ui/src/server/components/settings.ts

```diff
@@ -14,8 +14,8 @@ function getSettingsConfig(): Partial<Required<YTCoreConfig>['userSettingsConfig
     return userSettingsConfig || {};
 }
 
-function getSettingsSetup() {
-    return getRobotYTApiSetup(getSettingsConfig().cluster!).setup;
+function getSettingsSetup(cluster: string) {
+    return getRobotYTApiSetup(getSettingsConfig().cluster ?? cluster).setup;
```

Comment:
It is better to pass cluster to `getSettingsConfig(cluster)` and it would be also useful for `home.ts`.